activesupport/lib/active_support/file_evented_update_checker.rb

```diff
@@ -0,0 +1,72 @@
+require 'listen'
+
+module ActiveSupport
+  class FileEventedUpdateChecker
+    attr_reader :listener
+    def initialize(files, dirs={}, &block)
+      @files = Set.new
+      @files = files.map { |f| File.expand_path(f)}
+      @dirs = Hash.new
+      dirs.map do |key,value|
```

Comment:
Also, we should use a different name for the argument 'dirs' to prevent the shadow warning.